core/src/test/java/com/linecorp/armeria/server/HttpServiceTest.java

```diff
@@ -176,4 +192,23 @@ void contentLengthIsNotSetWhenTrailerExists() {
         assertThat(res.headers().get(HttpHeaderNames.CONTENT_LENGTH)).isNull();
         assertThat(res.trailers().get(HttpHeaderNames.of("foo"))).isEqualTo("baz");
     }
+
+    @ParameterizedTest
+    @EnumSource(value = SessionProtocol.class, names = {"H1C", "H2C"})
```

Comment:
Should we also check `ExchangeType` on the server side?

Reply:
I realized that the behavior differs between unary and streaming responses at the moment.
- Unary: A 500 internal exception is returned.
- Streaming: The connection is terminated immediately.

I've unified the behavior for both cases such that the connection is terminated immediately for consistency.
I doubt this is a normal path for users anyways.